packages/ui/src/Button.tsx

```diff
@@ -80,62 +83,78 @@ export type ButtonProps = ButtonCommonProps &
  * - **Primary**: Use primary button for the single primary action on the screen. To call attention to an action on a form, or highlight the strongest call to action on a page. Primary button should only appear once per screen. Not every screen requires a primary button.
  * - **Secondary**: Use secondary buttons for all remaining actions. Secondary button is the standard button for most use cases.
  */
-export const Button: FC<ButtonProps> = ({
-  kind = 'primary',
-  className,
-  disabled,
-  disabledTooltip,
-  children,
-  capitalize = true,
-  // Left icon
-  iconLeft,
-  iconLeftAriaLabel,
-  iconLeftSize,
-  iconLeftColor,
-  iconLeftClassName,
-  // Right icon
-  iconRight,
-  iconRightAriaLabel,
-  iconRightSize,
-  iconRightColor,
-  iconRightClassName,
-  ...rest
-}) => (
-  <Tooltip overlay={disabled ? disabledTooltip : undefined}>
-    <button
-      data-test="button"
-      className={cn(className, styles.button, {
-        [styles.primary]: kind === 'primary',
-        [styles.secondary]: kind === 'secondary',
-        [styles.transparent]: kind === 'transparent',
-      })}
-      disabled={disabled}
-      {...rest}
-    >
-      <span className={styles.outline} />
-      <span className={styles.body}>
-        {iconLeft && iconLeftAriaLabel && (
-          <Icon
-            icon={iconLeft}
-            ariaLabel={iconLeftAriaLabel}
-            data-test="button-icon-left"
-            className={cn(styles.iconLeft, iconLeftClassName)}
-            size={iconLeftSize}
-            color={iconLeftColor}
-          />
-        )}
-        <TruncatedText text={capitalize ? children.toUpperCase() : children} />
-        {iconRight && iconRightAriaLabel && (
-          <Icon
-            icon={iconRight}
-            ariaLabel={iconRightAriaLabel}
-            data-test="button-icon-right"
-            className={cn(styles.iconRight, iconRightClassName)}
-            size={iconRightSize}
-            color={iconRightColor}
-          />
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      kind = 'primary',
+      className,
+      disabled,
+      disabledTooltip,
+      disabledTooltipId,
+      children,
+      capitalize = true,
+      // Left icon
+      iconLeft,
+      iconLeftAriaLabel,
+      iconLeftSize,
+      iconLeftColor,
+      iconLeftClassName,
+      // Right icon
+      iconRight,
+      iconRightAriaLabel,
+      iconRightSize,
+      iconRightColor,
+      iconRightClassName,
+      ...rest
+    },
+    ref,
+  ) => {
+    /* Generate backup tooltip id if prop is empty */
+    const tooltipId = disabledTooltipId ?? `${uuid()}-button-tooltip`
```

Comment:
I am not sure we need it. If `disabledTooltipId` is missing, than the button will never have a tooltip. Should we change the props of the tooltip to:
```tsx
type TooltipContentProps = {
  content: ReactChild
  id: string
} | {
  content: undefined
  id: never
}
export type TooltipProps = {
  hideTimeoutDuration?: number
  offset?: number
  placement?: Placement
  visible?: boolean
  children: (ref: React.Dispatch<React.SetStateAction<HTMLElement | null>>) => ReactNode
}	} & TooltipContentProps
```
?